### DISCUSSION TITLE
System requirement upgrades for shopware 6.6

### DISCUSSION THREAD

**Discussion by keulinho** [shopware/platform#3359] [RFC / Proposal]
# Effort
Low

# Description
We want to upgrade the following system requirements with shopware v6.6.

## PHP
Upgrade from min 8.1 to **[8.2](https://php.watch/versions/8.2)**, this is basically implicit when we do the [Symfony 7 upgrade](https://github.com/shopware/shopware/discussions/3354), as they have the [same requirement](https://symfony.com/releases/7.0).
Additionally, it will allow us to use the new [PHP 8.2 features](https://php.watch/versions/8.2).

## Maria DB
Upgrade from min 10.3 to **10.11**, as it is the [latest LTS version](https://endoflife.date/mariadb) of Maria DB. Additionally, that version also supports the `JSON_OVERLAPS` function, that is needed by the "delete unused media" feature, so it finally will work in every setup.

## Redis
Upgrade from min 6.2 to **7.0**, as there were a lot of performance improvements in v7 and it allows us to use the new [redis function feature](https://redis.io/docs/interact/programmability/functions-intro/) in the future.

## Node
Upgrade from min 18 to **20**, as it is the [new LTS version](https://nodejs.dev/en/about/releases/) with active support until the end of 2024 and maintenance support until mid 2026-

# Benefits
Those upgrades allow us and the community to use the latest features of the tools and frameworks we depend on, this forms a solid foundation for the next major cycle. Also, all minimum versions offer long-term support, which should give us a stable and secure ground to build on.

**Comment by ener-Dev** [comment]
As of now, there is no support for MariaDB 10.11 under the CloudLinux module: DB Governor, as indicated by the changelog: https://changelog.cloudlinux.com/cl-MariaDB106. Under these circumstances, a technical upgrade is not feasible for us and would affect all of our shared hosting services.

The associated ticket is currently in the review stage.

**Reply by ener-Dev** [reply]
Comment from Cloudlinux: Our developers informed that the ETA is November 3rd.

**Reply by keulinho** [reply]
the 6.6 major release definitely won't be published this year, so I hope that shouldn't be an issue by the time the upgrade is required.

**Reply by ener-Dev** [reply]
We expect to start the upgrades at the end of January after the Christmas shopping season.
